src/utils.rs

```diff
@@ -0,0 +1,39 @@
+use std::fs::File;
+use std::path::Path;
+
+use serde::Serialize;
+
+pub fn write_json_to_file<T: Serialize, P: AsRef<Path>>(
+    obj: T,
+    path: P,
+) -> Result<(), std::io::Error> {
+    let mut file = File::create(path)?;
+    serde_json::to_writer(&mut file, &obj)?;
+    Ok(())
+}
+
+pub fn cleanup_tmp_files(tmp_dir: &tempfile::TempDir) {
+    if let Err(e) = std::fs::remove_dir_all(tmp_dir) {
+        eprintln!("Failed to clean up temporary directory: {}", e);
+    }
+}
+
+#[derive(serde::Deserialize)]
+pub struct Config {
+    download_dir: String,
+    #[allow(dead_code)]
+    url: String,
+    file_names: Vec<String>,
+    #[allow(dead_code)]
+    sha256_sums: Vec<String>,
+}
+
+pub fn set_env_vars(config: &[u8]) {
```

Comment:
I would keep the parsing of the config seperate from the setting of the env variables:

In the future you might want to have multiple functions taking the config and having to re-parse it in every such function is painful.

Reply:
Done